server/sheets.go

```diff
@@ -118,13 +118,28 @@ func prune(dirs []*Dir) []*Dir {
 	return dirs
 }
 
+func hasDeflate(acceptEncoding []string) bool {
+	for _, s := range acceptEncoding {
+		for _, encoding := range strings.Split(s, ", ") {
+			if encoding == "deflate" {
+				return true
+			}
+		}
+	}
+	return false
+}
+
 func (s *Server) sheetHandler(w http.ResponseWriter, r *http.Request) {
 	entity, access, ok := s.loadSheet(w, r)
 	if !ok {
 		return
 	}
 	response := sheet.NewSheetFromEntity(entity.Entity, entity.OriginalHash != entity.CurrentHash, access.ReadOnly)
-	CompressedJSONResponse(w, http.StatusOK, response)
+	if hasDeflate(r.Header.Values("Accept-Encoding")) {
```

Comment:
I think I'd prefer to have a function that does this so that the same if/else doesn't need to be repeated everywhere. Perhaps PossiblyCompressedJSONResponse()?